qiskit/pulse/transforms/channel_transforms.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import numpy as np
+from typing import Union
+
+from base_transforms import target_qobj_transform
+from qiskit.providers.ibmq import IBMQBackend
+from qiskit.pulse import Schedule, Play, MeasureChannel
+from qiskit.pulse.channels import PulseChannel
+from qiskit.visualization.pulse_v2.events import ChannelEvents
+
+def get_channel_waveform(sched: Schedule, 

```

Comment:
Should this also accept a `ScheduleBlock` @nkanazawa1989?

Reply:
Not needed if it is correctly implemented. This transform pass internally calls full `target_qobj_transform`, however, we need a specific transform function that takes a low-level representation (all directives are dropped, contexts are resolved, subroutine is flattened, etc...) and generate a raw waveform data by parsing frames. In this case, the the input program should be `Schedule`.

Reply:
The `target_qobj_transform` function has the input annotation as `sched: Union[ScheduleBlock, Schedule, InstructionSched, Iterable[InstructionSched]]`. Maybe I should just do the same here?